### DISCUSSION TITLE
Ace not present in build folder

### DISCUSSION THREAD

**Discussion by bitkidd** [adonisjs/core#2566] [Help] [answered]
Initiated a new app and tried to build it via `node ace build --production`, the `build` folder appeared but no `ace` there, as running a command `node build/ace migration:run --force` leads to an error:
```bash
internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module '***/build/ace'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

Actual `package.json` deps:
```json
"devDependencies": {
    "@adonisjs/assembler": "^5.1.1",
    "adonis-preset-ts": "^2.1.0",
    "eslint": "^7.25.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-adonis": "^1.3.1",
    "eslint-plugin-prettier": "^3.4.0",
    "pino-pretty": "^4.7.1",
    "prettier": "^2.2.1",
    "typescript": "~4.2",
    "youch": "^2.2.2",
    "youch-terminal": "^1.1.1"
  },
  "dependencies": {
    "@adonisjs/core": "^5.1.6",
    "@adonisjs/lucid": "^14.0.0",
    "@adonisjs/repl": "^3.1.2",
    "luxon": "^1.26.0",
    "pg": "^8.6.0",
    "proxy-addr": "^2.0.6",
    "reflect-metadata": "^0.1.13",
    "source-map-support": "^0.5.19",
    "uuid": "^8.3.2"
  }
```

**Comment by bitkidd** [comment]
Rolled back to `"@adonisjs/core": "^5.0.4-preview-rc-2.1"` and `"@adonisjs/assembler": "^3.1.1",`, build runs fine.

**Comment by bitkidd** [comment, marked as answer]
Okay, found the answer. I had an unused variable, that led to TS error, that led to build not being completed. Using the flag `--ignore-ts-errors` resolved the problem.

**Reply by genu** [reply]
This causes a failure in production I think (Heroku in my case). The linter also doesn't catch it either. 🤔  

Need to adress this with eslint or prettier. Or offer some guidenance.

**Reply by bitkidd** [reply]
Run the build locally `node ace build --production` to check the errors and fix them before pushing to production server.
